### PR TITLE
improve docker build

### PR DESCRIPTION
**Pull request by inductor**
Addressed to this: https://github.com/mxplusb/cloudflare-sync/issues/7 and better docker building strategy using multi-stage build

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -113,8 +113,6 @@ func main() {
 	go ipy.updateCloudflare()
 
 	logger.Println("workers booted.")
-
-	<-quit
```
Comment (siennathesane) [inline main.go:117]: I thought I had caught all of them, I should prolly add some CI to this.

**Review by siennathesane — CHANGES_REQUESTED** [review]
I'd like to make changes. I'm pasting them here for easier reading instead of individual comments for ease of reading.

```
FROM golang:alpine as builder # <- remove version for easier maintenance.
# ENV CGO_ENABLED=0 GOOS=linux GOARCH=amd64 # <- not needed
WORKDIR /go/src/cloudflare-sync # <- this doesn't need to be changed there is module support in all supported versions of go
# COPY go.mod go.sum ./ <- not needed
# RUN GO111MODULE=on go mod download <- also not needed
```

**Comment by inductor** [comment]
@mxplusb I believe you at least need the minor version stability for the base image, unless you actually make tests or at least CI environment. What do you think?

**Comment by inductor** [comment]
Also I'm explaining this for you just in case because the following lines are especially for `Dockerfile` design wise purpose and you might miss what I did here.

```Dockerfile
# If this is in the cache, the go mod download part will be ignored
COPY go.mod go.sum ./
RUN GO111MODULE=on go mod download

# This will not be as-is since most cases you make changes in source code
COPY . .
RUN go build -o cloudflare-sync
```

These lines should be important because go.mod and go.sum is less frequently changed than the source code lines. That means  by separating these works for better caching of Docker image layers.

**Comment by siennathesane** [comment]
@inductor thanks for your contribution, sorry for letting it languish, life got busy. I don't think I'm going to accept this change. The docker build as it exists right now is simple and works, so I'm going to leave it for now.

Please don't let this prevent you from contributing more, I'm just not interested in this specific change. Thank you, though!

**Pull request closed without merging**